File: Backend/Modelos/usuario.py
from pydantic import EmailStr, Field
from typing import Optional
from Modelos.modeloBase import ModeloBase

# Crear usuarios
class UsuarioCreacion(ModeloBase):
    correo: EmailStr
    contrasenia: str = Field(..., min_length=6)
    nombreUsuario: str = Field(..., min_length=3, max_length=40)
    descripcion: Optional[str] = "Me encanta esta aplicacion, es fantastica"
    fotoPerfil: Optional[str] = "default.png"

# Devuelve el servidor
class UsuarioPublico(ModeloBase):
    id: str = Field(..., alias="_id") # Mapear el _id de Mongo a "id" en base.py
    correo: EmailStr
    nombreUsuario: str
    descripcion: str
    fotoPerfil: str
    calificacionMedia: Optional[float] = Field(None, ge=1.0, le=5.0)
    rol: str

class UsuarioEdicion(ModeloBase):
    nombreUsuario: Optional[str] = Field(None, min_length=3, max_length=40)
    descripcion: Optional[str] = None
    fotoPerfil: Optional[str] = None

class Valoracion(ModeloBase):
    puntuacion: float = Field(..., ge=1.0, le=5.0)

File: Frontend/AppFavoresEscritorio/AppFavoresEscritorio/Modelos/Autenticacion/Usuario.cs
using System.Text.Json.Serialization;

namespace AppFavoresEscritorio.Modelos.Autenticacion;

public class UsuarioCreacion
{
    [JsonPropertyName("correo")]
    public string Correo { get; set; } = string.Empty;

    [JsonPropertyName("contrasenia")]
    public string Contrasenia { get; set; } = string.Empty;

    [JsonPropertyName("nombreUsuario")]
    public string NombreUsuario { get; set; } = string.Empty;

    [JsonPropertyName("descripcion")]
    public string Descripcion { get; set; } = "Me encanta esta aplicacion";

    [JsonPropertyName("fotoPerfil")]
    public string FotoPerfil { get; set; } = "default.png";
}
